src/stores/wordpress-versions-slice.ts

```diff
@@ -0,0 +1,82 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+
+const MINIMUM_WORDPRESS_VERSION = '5.9.9';
+
+interface WordPressVersion {
+	version: string;
+	isBeta: boolean;
+}
+
+interface WordPressOffer {
+	version: string;
+	response: 'autoupdate' | 'upgrade';
+}
+
+interface WordPressApiResponse {
+	offers: WordPressOffer[];
+}
+
+interface WordPressVersionsState {
+	versions: WordPressVersion[];
+	status: 'idle' | 'loading' | 'succeeded' | 'failed';
+	error: string | null;
+}
+
+const initialState: WordPressVersionsState = {
+	versions: [],
+	status: 'idle',
+	error: null,
+};
+
+export const fetchWordPressVersions = createAsyncThunk(
+	'wordpressVersions/fetchWordPressVersions',
+	async () => {
+		const response = await fetch(
+			`https://api.wordpress.org/core/version-check/1.7/?channel=beta&version=${ MINIMUM_WORDPRESS_VERSION }`
```

Comment:
Would it make sense to extract code responsible for the API call to `src/api` according to the [Project Structure](https://github.com/Automattic/studio/blob/trunk/docs/code-contributions.md#project-structure) doc?

Reply:
We could, but my two cents is that for such a simple network request, we would make things unnecessarily complex by moving it to a utility function. 

As a general comment on the `src/api` folder, I think we'll find that requesting WordPress.com APIs from hooks or Redux slices will be much easier because we can access the WPCOM client there (not yet in Redux stores, but we will be able to once `AuthProvider` has been refactored to a Redux store).

Reply:
In such a case, would we store all API calls along with Redux stores and skip using `src/api` completely?

Reply:
I like the idea of keeping the API calls inside the thunks 👍